### PR TITLE
New options to Interface 

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
New Options for Interface 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
- na_ontap_interface.py 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by carchi8py** [comment]
@thedoubl3j how do we force ansibot to return this?

**Review by thedoubl3j — COMMENTED** [review]
docs look clear, just need CI fixed.

File: lib/ansible/modules/storage/netapp/na_ontap_interface.py [hunk 1]
```diff
@@ -260,6 +294,12 @@ def set_options(options, parameters):
             options['administrative-status'] = parameters['admin_status']
         if parameters.get('force_subnet_association') is not None:
             options['force-subnet-association'] = 'true' if parameters['force_subnet_association'] else 'false'
+        if parameters.get('dns_domain_name') is not None:
+            options['dns-domain-name'] = parameters['dns_domain_name']
+        if parameters.get('listen_for_dns_query') is not None:
+            options['listen-for-dns-query'] = str(parameters['listen_for_dns_query'])
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_interface.py:300]: Is it enough to use str, or do we need to convert to lowercase?

Look also at line 296.  Should we use the same construct?

File: lib/ansible/modules/storage/netapp/na_ontap_interface.py [hunk 2]
```diff
@@ -237,6 +263,14 @@ def get_interface(self):
                 return_value['netmask'] = interface_attributes['netmask']
             if interface_attributes.get_child_by_name('firewall-policy'):
                 return_value['firewall_policy'] = interface_attributes['firewall-policy']
+            if interface_attributes.get_child_by_name('dns-domain-name') is not 'none':
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_interface.py]: should be   != 'none'

is not None is correct.  is not 'none' is not correct.

**Review by lonico — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/na_ontap_interface.py (hunk 1 above)
Comment (lonico) [reply lib/ansible/modules/storage/netapp/na_ontap_interface.py:300]: I confirmed the ZAPI accepts both False and false, or True and true, so this is OK as is.

It's only when making comparisons with returned values (get), that we need to be case sensitive.

**Comment by thedoubl3j** [comment]
@carchi8py @lonico still waiting on fixes for this one

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
lgtm, tests updated with new features. shipit

**Pull request merged**